DRILL-6744: Support varchar and decimal push down

arina-ielchiieva opened the pull request:
1. Added enableStringsSignedMinMax parquet format plugin config and store.parquet.reader.strings_signed_min_max session option to control reading binary statistics for files generated by prior versions of Parquet 1.10.0.
2. Added ParquetReaderConfig to store configuration needed during reading parquet statistics or files.
3. Provided mechanism to enable varchar / decimal filter push down.
4. Added VersionUtil to compare Drill versions in string representation.
5. Added appropriate unit tests.

Details in [DRILL-6744](https://issues.apache.org/jira/browse/DRILL-6744).

arina-ielchiieva commented:
@vvysotskyi please review.

vvysotskyi commented on exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/ParquetReaderConfig.java:
```diff
@@ -0,0 +1,175 @@
+ * Should be initialized using {@link Builder}, constructor is made public only for ser / de purposes.
+ */
+@JsonInclude(JsonInclude.Include.NON_DEFAULT)
+public class ParquetReaderConfig {
```
Is it possible to make this class immutable, so in `getDefaultInstance()` method we will be able to use the same object instead of instantiating the new one?

vvysotskyi commented on exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/stat/ParquetMetaStatCollector.java:
```diff
@@ -132,62 +129,163 @@ public ParquetMetaStatCollector(ParquetTableMetadataBase parquetTableMetadata,
+                  minBytes = new BigInteger((byte[]) min).toByteArray();
+                  maxBytes = new BigInteger((byte[]) max).toByteArray();
+                }
+                break;
```
`break` and `default` here may be removed.

vvysotskyi commented on exec/java-exec/src/test/java/org/apache/drill/exec/store/parquet/TestParquetReaderConfig.java:
```diff
@@ -0,0 +1,122 @@
+
+  @Test
+  public void testReadOptions() {
+    ParquetReaderConfig readerConfig = new ParquetReaderConfig();
```
Looks like we contravene the recommendation from its Javadoc

arina-ielchiieva commented:
@vvysotskyi addressed code review comments:
1. Used VersionUtil from Hadoop lib.
2. Made ParquetReaderConfig immutable.
3. Added trace login instead of removing default in switch.
4. Made other changes as requested.

Thanks for the code review.
